### PR TITLE
Update Leaflet, D3, and jQuery dependencies

### PR DESCRIPTION
**Pull request by jwasilgeo**
Hi @beuan, as requested, this PR updates 3rd party library dependencies, including:

- Leaflet
  - I tried to test upgrading to a newer leaflet.pm plugin (now geoman-io) that included the hoped for fixes @beuan and @jgravois wanted to have, but it didn't work as a drop-in upgrade. Perhaps not worth it for now. Comments were added to cover these findings.
- D3
  - Change to CDN recommended by [their docs](https://d3js.org/getting-started#d3-in-vanilla-html).
  - Remove some unused pieces.
- jQuery
  - Switch to `min` version of their JS.
  - At this time I don't think we can effectively remove it unless we first move away from using JQuery UI.

### CODE REVIEW

File: previewMap.js [hunk 1]
```diff
@@ -222,7 +222,7 @@ function pickProjection(lat0, lon0, projectionString) {
 			.precision(.1);
 	}
 	else if (projectionString == 'Natural Earth') {
-		return d3.geoNaturalEarth()
+		return d3.geoNaturalEarth1() // TODO: or d3.geoNaturalEarth2() ?
```
Comment (jwasilgeo) [inline previewMap.js]: cc @beuan

File: index.html [hunk 2]
```diff
@@ -12,29 +12,37 @@
...
-		<script src="https://d3js.org/d3-array.v1.min.js"></script>
-		<script src="https://d3js.org/d3-fetch.v1.min.js"></script>
-		<script src="https://d3js.org/d3-geo.v1.min.js"></script>
-		<script src="https://d3js.org/d3-geo-projection.v2.min.js"></script>
-		<script src="lib/d3-geo-polygon.min.js"></script>
-		<script src="https://unpkg.com/topojson-client@3"></script>
+		<script src="https://cdn.jsdelivr.net/npm/d3@7"></script>
+		<script src="https://cdn.jsdelivr.net/npm/d3-geo@3"></script>
+		<script src="https://cdn.jsdelivr.net/npm/d3-geo-projection@4"></script>
+		<script src="https://cdn.jsdelivr.net/npm/d3-geo-polygon@1"></script>
```
Comment (jwasilgeo) [inline index.html:34]: Info: `d3-geo-polygon` version 2 crashes with two-point equidistant, so I'll keep it at v1 for now.

File: index.html [hunk 3]
```diff
@@ -12,29 +12,37 @@
...
+     		integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
+     		crossorigin=""></script>
+		<!-- TODO: 
+		 Verify this is fixed as of https://github.com/geoman-io/leaflet-geoman/releases/tag/2.8.0
+		 via https://github.com/geoman-io/leaflet-geoman/pull/663
+		 via https://github.com/geoman-io/leaflet-geoman/pull/489
+		 which is all related to what we need in: https://github.com/geoman-io/leaflet-geoman/pull/464
+
+		 If fixed, we should be able to remove our local patch and go to version 2.8.0 for CSS and JS and test it out.
+		-->
```
Comment (jwasilgeo) [inline index.html]: As noted in the PR description, likely it is lowest friction and effort to leave all this alone. If that's OK, I can remove these comments.

File: index.html (hunk 2 above)
Comment (beuan) [reply index.html:34]: @jwasilgeo, sounds good. Thank you!

File: previewMap.js (hunk 1 above)
Comment (beuan) [reply previewMap.js]: @jwasilgeo
Natural Earth 1 is correct.

File: index.html (hunk 3 above)
Comment (beuan) [reply index.html]: Thank you, @jwasilgeo! I think should be OK and I remove the comment before I merge it.

**Review by beuan — APPROVED** [review]

**Comment by beuan** [comment]
Thank you, @jwasilgeo, for doing all this! Very much appreciated!
I merged this changes to my update branch, so that I can install everything together.

**Comment by jwasilgeo** [comment]
Wonderful!

**Pull request merged**